src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -674,11 +683,11 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
 
                 if let Some(item) = assoc_item {
                     let kind = item.kind;
-                    let fragment = UrlFragment::from_assoc_item(item_name, kind, false);
+                    let fragment = UrlFragment::from_assoc_item(item.def_id, kind, false);
                     // HACK(jynelson): `clean` expects the type, not the associated item
                     // but the disambiguator logic expects the associated item.
                     // Store the kind in a side channel so that only the disambiguator logic looks at it.
```

Comment:
Does this means that this comment is now no longer relevant ? If so does this means that the json output now report the associated item and not the type ?

Reply:
Yes to the first part. For the second part, I'm not sure. Probably it's the same.